integration/tests/access/cohort4/websocket_subscriptions_test.go

```diff
@@ -0,0 +1,1065 @@
+		testnet.WithAdditionalFlag("--execution-data-indexing-enabled=true"),
+		testnet.WithAdditionalFlagf("--execution-state-dir=%s", testnet.DefaultExecutionStateDir),
+		testnet.WithAdditionalFlagf("--websocket-inactivity-timeout=%ds", InactivityTimeout),
+		testnet.WithAdditionalFlagf("--websocket-max-subscriptions-per-connection=%d", MaxSubscriptionsPerConnection),
+		testnet.WithAdditionalFlagf("--experimental-enable-websockets-stream-api=true"),
+		testnet.WithMetricsServer(),
+	)
+
+	// add the ghost (access) node config
+	ghostNode := testnet.NewNodeConfig(
+		flow.RoleAccess,
+		testnet.WithLogLevel(zerolog.FatalLevel),
+		testnet.AsGhost())
+
+	consensusConfigs := []func(config *testnet.NodeConfig){
+		testnet.WithAdditionalFlag("--cruise-ctl-fallback-proposal-duration=100ms"),
+		testnet.WithAdditionalFlag(fmt.Sprintf("--required-verification-seal-approvals=%d", 1)),
+		testnet.WithAdditionalFlag(fmt.Sprintf("--required-construction-seal-approvals=%d", 1)),
+		testnet.WithLogLevel(zerolog.FatalLevel),
+	}
+
+	nodeConfigs := []testnet.NodeConfig{
+		testnet.NewNodeConfig(flow.RoleCollection, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleCollection, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleExecution, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleExecution, testnet.WithLogLevel(zerolog.FatalLevel)),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleConsensus, consensusConfigs...),
+		testnet.NewNodeConfig(flow.RoleVerification, testnet.WithLogLevel(zerolog.FatalLevel)),
+		bridgeANConfig,
+		ghostNode,
+	}
+
+	conf := testnet.NewNetworkConfig("websockets_subscriptions_test", nodeConfigs)
+	s.net = testnet.PrepareFlowNetwork(s.T(), conf, flow.Localnet)
+
+	// start the network
+	s.T().Logf("starting flow network with docker containers")
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+
+	s.net.Start(s.ctx)
+
+	accessUrl := fmt.Sprintf("localhost:%s", s.net.ContainerByName(testnet.PrimaryAN).Port(testnet.GRPCPort))
+	var err error
+	s.grpcClient, err = common.GetAccessAPIClient(accessUrl)
+	s.Require().NoError(err)
+
+	s.serviceClient, err = s.net.ContainerByName(testnet.PrimaryAN).TestnetClient()
+	s.Require().NoError(err)
+
+	s.restAccessAddress = s.net.ContainerByName(testnet.PrimaryAN).Addr(testnet.RESTPort)
+
+	// pause until the network is progressing
+	var header *sdk.BlockHeader
+	s.Require().Eventually(func() bool {
+		header, err = s.serviceClient.GetLatestSealedBlockHeader(s.ctx)
+		s.Require().NoError(err)
+
+		return header.Height > 0
+	}, 30*time.Second, 1*time.Second)
+}
+
+// TestInactivityTracker tests that the WebSocket connection closes due to inactivity
+// after the specified timeout duration.
+func (s *WebsocketSubscriptionSuite) TestInactivityTracker() {
```

Comment:
integration test setup is heavy since it's starting lots of containers. When the test configuration is the same, for several test cases, you can reuse the same network across multiple tests.

one way to do this is to have a top level `Test*` method, then have it call several `test*` methods that do individual tests. This makes the suites run a lot faster in CI and tend to be less flaky.

so for example, you may have a test `TestWebsocketSubscriptions`
which calls
* `s.testInactivityTracker`
* `s.testMaxSubscriptionsPerConnection`
* etc